commit bd1ba664147ae271e592aa90266cf9aba6efe116
Author: Johannes Berg <johannes.berg@intel.com>
Date:   Thu Nov 13 20:53:45 2014 +0100

    iwlwifi: mvm: move U-APSD decision to authentication
    
    In order to change the usage of U-APSD on the fly later,
    move the enabling condition into a new function that is
    called when authenticated.
    
    This allows the module parameter to become writable, it
    won't take effect immediately but at least on the next
    association the new value will be used.
    
    Signed-off-by: Johannes Berg <johannes.berg@intel.com>
    Signed-off-by: Emmanuel Grumbach <emmanuel.grumbach@intel.com>

diff --git a/drivers/net/wireless/iwlwifi/iwl-drv.c b/drivers/net/wireless/iwlwifi/iwl-drv.c
index 0d3472679e4d..e3af724e0e0a 100644
--- a/drivers/net/wireless/iwlwifi/iwl-drv.c
+++ b/drivers/net/wireless/iwlwifi/iwl-drv.c
@@ -1490,7 +1490,7 @@ module_param_named(d0i3_disable, iwlwifi_mod_params.d0i3_disable,
 MODULE_PARM_DESC(d0i3_disable, "disable d0i3 functionality (default: Y)");
 
 module_param_named(uapsd_disable, iwlwifi_mod_params.uapsd_disable,
-		   bool, S_IRUGO);
+		   bool, S_IRUGO | S_IWUSR);
 #ifdef CONFIG_IWLWIFI_UAPSD
 MODULE_PARM_DESC(uapsd_disable, "disable U-APSD functionality (default: N)");
 #else